### PR TITLE
Ported Halt Play Test and Stop Play Test to ER Force Simulator

### PR DESCRIPTION
**Pull request by Joshuayellowley**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 1]
```diff
@@ -83,14 +85,14 @@ TEST_F(StopPlayTest, test_stop_play_friendly_half_corner_robots_close_together)
 {
     BallState ball_state(Point(-4, -2.5), Vector(0, 0));
     auto friendly_robots = TestUtil::createStationaryRobotStatesWithId(
-        {Point(-3, -2.5), Point(-4, -2), Point(-2, -2.5), Point(-3, -2), Point(-3.5, -2),
+        {Point(-2, -2.5), Point(-4, -2), Point(-2, -2.5), Point(-3, -2), Point(-3.5, -2),
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/stop_play_test.cpp]: is this required for the test to pass?

**Review by jonathanlew — COMMENTED** [review]
looks good just one question

File: src/software/ai/hl/stp/play/stop_play_test.cpp (hunk 1 above)
Comment (Joshuayellowley) [reply src/software/ai/hl/stp/play/stop_play_test.cpp]: Just disabled the test given our discussion last weekend

File: src/software/simulated_tests/non_terminating_validation_functions/robots_avoid_ball_validation.cpp [hunk 2]
```diff
@@ -7,6 +7,8 @@ void robotsAvoidBall(double min_distance, std::vector<RobotId> excluded_robots,
                      std::shared_ptr<World> world_ptr,
                      ValidationCoroutine::push_type& yield)
 {
+    std::cout << "BALL x: " << world_ptr->ball().position().x()
+              << " y: " << world_ptr->ball().position().y() << std::endl;
```
Comment (jonathanlew) [inline src/software/simulated_tests/non_terminating_validation_functions/robots_avoid_ball_validation.cpp]: debug code

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 3]
```diff
@@ -74,12 +75,14 @@ TEST_F(StopPlayTest, test_stop_play_friendly_half_robots_spread_out)
...
         stop_play_rules;
 
-    runTest(field, ball_state, friendly_robots, enemy_robots,
+    runTest(field_type, ball_state, friendly_robots, enemy_robots,
             terminating_validation_functions, non_terminating_validation_functions,
             Duration::fromSeconds(10));
 }
 
-TEST_F(StopPlayTest, test_stop_play_friendly_half_corner_robots_close_together)
+// TODO: Re-enable when test is fixed
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/stop_play_test.cpp]: please attach an issue

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 4]
```diff
@@ -4,26 +4,26 @@
...
     std::vector<ValidationFunction> stop_play_rules;
 
-
     std::vector<ValidationFunction> initStopPlayRules()
     {
         return {
             [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
                 // Wait 2 seconds for robots that start too close to the ball to move away
-                if (world_ptr->getMostRecentTimestamp() >= Timestamp::fromSeconds(2))
+                // EDIT: Updated to 4 seconds for Er Force Simulator
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/stop_play_test.cpp]: please just update the comment above without the edit

**Review by jonathanlew — APPROVED** [review]
LGTM!

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 5]
```diff
@@ -74,12 +75,14 @@ TEST_F(StopPlayTest, test_stop_play_friendly_half_robots_spread_out)
...
-    runTest(field, ball_state, friendly_robots, enemy_robots,
+    runTest(field_type, ball_state, friendly_robots, enemy_robots,
             terminating_validation_functions, non_terminating_validation_functions,
             Duration::fromSeconds(10));
 }
 
-TEST_F(StopPlayTest, test_stop_play_friendly_half_corner_robots_close_together)
+// TODO (#2544): Re-enable when test is fixed
+// Disabled when porting to ER_force simulator
+TEST_F(StopPlayTest, DISABLED_test_stop_play_friendly_half_corner_robots_close_together)
```
Comment (nimazareian) [inline src/software/ai/hl/stp/play/stop_play_test.cpp:85]: The current bug causing this one to fail is related to the path planner not being able to find a path. @jonathanlew maybe you could try reenabling this in your upcoming PR with the new GlobalPathPlanner

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 6]
```diff
@@ -174,7 +177,7 @@ TEST_F(StopPlayTest, DISABLED_test_stop_play_ball_in_front_of_friendly_defense_a
     std::vector<ValidationFunction> non_terminating_validation_functions =
         stop_play_rules;
 
-    runTest(field, ball_state, friendly_robots, enemy_robots,
```
Comment (nimazareian) [inline src/software/ai/hl/stp/play/stop_play_test.cpp:177]: This test (i.e. `test_stop_play_ball_in_front_of_friendly_defense_area`) is resulting in a SIGSEGV for me, though the comment regarding the bug also seems to be related to path planner. @jonathanlew maybe you can try seeing if this gets fixed in your PR aswell.

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 7]
```diff
@@ -4,26 +4,26 @@
...
-
     std::vector<ValidationFunction> initStopPlayRules()
     {
         return {
             [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-                // Wait 2 seconds for robots that start too close to the ball to move away
-                if (world_ptr->getMostRecentTimestamp() >= Timestamp::fromSeconds(2))
+                // Wait 4 seconds for robots that start too close to the ball to move away
+                // with ER Simulator
+                if (world_ptr->getMostRecentTimestamp() >= Timestamp::fromSeconds(4))
```
Comment (nimazareian) [inline src/software/ai/hl/stp/play/stop_play_test.cpp:26]: Although the validation for this test is passing now, the behaviour of the play is still broken. However, #2571 seems to fix it when I tried it locally.
Current behaviour: 
halt_play_test_bug

**Comment by jonathanlew** [comment]
I was running into a SIGFPE with the box2d simulator so I had to port these tests over in #2582, so closing due to being superseded

**Pull request closed without merging**